src/CachePeer.cc

```diff
@@ -114,6 +118,209 @@ CachePeer::connectTimeout() const
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    // else default connection-auth=auto
+
+    secure.dumpCfg(os, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& yaml)
+{
+    yaml <<
+         spaces(2) << "- name: " << name << "\n" <<
+         spaces(4) << "type: " << typeString() << '\n' <<
+         spaces(4) << "http address: " << host << ':' << http_port << '\n';
+    if (icp.port)
+        yaml << spaces(4) << "icp address:" << host << ':' << icp.port << '\n';
+    yaml << spaces(4) << "options:";
+    dumpOptions(yaml);
+    yaml << '\n';
+
+    std::vector<Ip::Address> addr;
+    std::copy(addresses, addresses+n_addresses, std::back_inserter(addr));
+    yaml << spaces(4) << "addresses: [ " << AsList(addr).quoted().delimitedBy(", ") << " ]\n";
+
+    yaml <<
+         spaces(4) << "status: " << (neighborUp(this) ? "Up" : "Down") << '\n' <<
+         spaces(4) << "fetches: " << stats.fetches << '\n' <<
+         spaces(4) << "open connections: " << stats.conn_open << '\n' <<
+         spaces(4) << "average RTT: " << stats.rtt << " msec\n";
+
+    if (stats.last_query > 0) {
+        yaml << spaces(4) << "last query:" <<
+             (squid_curtime - stats.last_query) << " seconds ago\n";
+    }
+
+    if (stats.last_reply > 0) {
+        yaml << spaces(4) << "last reply: " <<
+             (squid_curtime - stats.last_reply) << " seconds ago\n";
+    }
+
+    yaml << spaces(4) << "pings sent: " << stats.pings_sent << '\n' <<
+         spaces(4) << "pings acked: " << stats.pings_acked << " " <<
+         Math::intPercent(stats.pings_acked, stats.pings_sent) << "%\n";
+
+    yaml << spaces(4) << "replies ignored: " << stats.ignored_replies << " " <<
+         Math::intPercent(stats.ignored_replies, stats.pings_acked) << "%\n";
+
+    auto sectionHeader = AtMostOnce("histogram of pings acked:\n");
+
+#if USE_HTCP
+    if (options.htcp) {
+        yaml << spaces(4) << sectionHeader;
+        yaml << spaces(6) << "htcp misses: " << htcp.counts[0] << " " <<
+             Math::intPercent(htcp.counts[0], stats.pings_acked) << "%\n" <<
+             spaces(6) << "htcp hits: " << htcp.counts[1] << " " <<
+             Math::intPercent(htcp.counts[1], stats.pings_acked) << "%\n";
+    } else {
+#endif
+        for (auto op : WholeEnum<icp_opcode>()) {
+            if (icp.counts[op] == 0)
+                continue;
+
+            yaml << spaces(4) << sectionHeader;
+            yaml << spaces(6) <<
+                 "icp " << icp_opcode_str[op] << ": " <<
```

Comment:
Why should this PR add an `icp ` prefix to icp_opcode_str strings that all start with an `ICP_` prefix?

Reply:
Thanks, removing.